<?xml version="1.0"?>
<launch>
  <rosparam param="/path_planning/inflation_radius">0.35</rosparam>
  <rosparam param="/path_planning/cost_radius">0.4</rosparam>
  <rosparam param="/path_planning/smoothing_alpha">0.9</rosparam>
  <rosparam param="/path_planning/smoothing_beta">0.05</rosparam>
  
  <arg name="robot_name" default="$(optenv ROBOT_NAME hsrb)"/>
  <include file="$(find hsrb_gazebo_launch)/launch/include/hsrb_gazebo_common.xml" pass_all_args="true" >
    <arg name="map" value="$(find bring_up)/maps/moon_station.yaml" />
    <arg name="world_name" value="$(find bring_up)/worlds/moon_station.world"/>
    <arg name="robot_pos" value="-x 0 -y 0 -z 0 -Y 0" />
    <arg name="use_laser_odom" value="false" />
    <arg name="use_manipulation" value="false" />
    <arg name="use_navigation" value="false" />
    <arg name="use_perception" value="false" />
    <arg name="use_task" value="false" />
    <arg name="use_teleop" value="false" />
    <arg name="use_web" value="false" />
    <arg name="use_laser_odom" value="false" />
    <arg name="rviz" value="false"/>
  </include>
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find etapa04)/src/rviz/mapping.rviz"/>
  <node name="gmapping" pkg="gmapping" type="slam_gmapping" args="scan:=/hsrb/base_scan" output="screen">
    <param name="map_update_interval" value="1.0" />
    <param name="maxUrange" value="10.4"/>
    <param name="linearUpdate" value="0.3"/>
    <param name="angularUpdate" value="0.2"/>
    <param name="xmin" value="-20"/>
    <param name="ymin" value="-20"/>
    <param name="xmax" value="20"/>
    <param name="ymax" value="20"/>
    <param name="maxRange" value="10.5"/>
    <param name="minimumScore" value="0.0"/>
  </node>
  
  <node name="periodic_mapping" pkg="etapa04" type="periodic_mapping.py"/>
  <node name="a_star" pkg="etapa04" type="a_star.py"/>
  <node name="path_smoothing" pkg="etapa04" type="path_smoothing.py"/>
  
  <node name="close_collision_node" pkg="etapa04" type="close_collision_ind.py"/>
  <node name="stuck_node" pkg="etapa04" type="stuck_ind.py"/>
  
  <node name="path_following" pkg="etapa04" type="path_following.py"/>
  <node name="pot_fields" pkg="etapa04" type="pot_fields.py"/>
  
  <node name="etapa04_node" pkg="etapa04" type="etapa04_sm.py"/>
  <!--
  
  
  <node name="periodic_mapping" pkg="etapa04" type="periodic_mapping.py"/>
  <node name="a_star" pkg="etapa04" type="a_star.py"/>
  <node name="path_smoothing" pkg="etapa04" type="path_smoothing.py"/>
  -->
  
  <!--

  <node name="map_saver" pkg="map_server" type="map_saver" args="-f current_map"/>
  <node name="map_server" pkg="map_server" type="map_server" output="screen" args="$(find bring_up)/maps/moon_station.yaml"/>
  -->
  
  <!--
  <node name="amcl" pkg="amcl" type="amcl" output="screen" args="scan:=/hsrb/base_scan">
    <param name="initial_pose_x" value="0.0"/>
    <param name="initial_pose_y" value="0.0"/>
    <param name="initial_pose_a" value="0.0"/>
  </node>
  -->
</launch>

